server/src/com/thoughtworks/go/server/materials/MaterialDatabaseUpdater.java

```diff
@@ -87,7 +81,7 @@ public void updateMaterial(final Material material) throws Exception {
                     LOGGER.debug(String.format("[Material Update] Material repository not found, creating with latest revision from %s", material));
                 }
 
-                synchronized (cacheKeyForMaterial) {
+                synchronized (materialMutex) {
```

Comment:
With the way ```MaterialUpdateService.updateMaterial``` is implemented there shouldn't be simultaneous update for the same material. And this behaviour does not differ post my changes. 
Was not sure which scenario this synchronisation handled, so retained it.